src/cljs/kixi/hecuba/tabs/hierarchy/xyplot.cljs

```diff
@@ -0,0 +1,354 @@
+             (raw-data/sorting-th owner "Resolution" :resolution)
+             (raw-data/sorting-th owner "Device ID" :device_id)
+             (raw-data/sorting-th owner "Location" :location)
+             (raw-data/sorting-th owner "Earliest Event" :lower_ts)
+             (raw-data/sorting-th owner "Last Event" :upper_ts)
+             (raw-data/sorting-th owner "Status" :status)]]
+           [:tbody
+            (for [row (if sort-asc
+                        (sort-by sort-key sensors-with-data)
+                        (reverse (sort-by sort-key sensors-with-data)))]
+              (om/build sensor-row {:sensor row :xyplot-data (:xyplot-data cursor)} {:opts {:event-chan event-chan}}))]]])))))
+
+(defn calc-linear-regression-line
+  [data]
+  (let [n (->> data count)
+        sum #(reduce + %)
+        sum-x (->> data (map first) sum)
+        sum-y (->> data (map second)  sum)
+        sum-x-times-y (->> data (map (fn [[x y]] (* x y))) sum)
+        sum-x-times-x (->> data (map (fn [[x y]] (* x x))) sum)
+        a (/ (- (* sum-y sum-x-times-x) (* sum-x sum-x-times-y))
+             (- (* n sum-x-times-x) (* sum-x sum-x)))
+        b (/ (- (* n sum-x-times-y) (* sum-x sum-y))
+             (- (* n sum-x-times-x) (* sum-x sum-x)))
+        fn #(float (+ a (* b %)))
+        min-x (->> data (map first) (apply min))
+        max-x (->> data (map first) (apply max))
+        xy1 [min-x (fn min-x)]
+        xy2 [max-x (fn max-x)]]
+    [xy1 xy2]))
+
+(defn process-data
+  [[x-data y-data]]
+  (let [x-raw-data (->> x-data (sort-by :timestamp) (mapv :value))
+        y-raw-data (->> y-data (sort-by :timestamp) (mapv :value))]
+    (if (not= (count x-raw-data) (count y-raw-data))
+      (throw (js/Error. "X and Y have different amounts of data (" (count x-raw-data) "/" (count y-raw-data) ")")))
+    (let [data {:values (mapv (fn [i] [(nth x-raw-data i) (nth y-raw-data i)]) (range 0 (count x-raw-data)))}
+          reg-line (calc-linear-regression-line (:values data))
+          data (assoc data :line reg-line)
+          x-range [(->> data :values (map first) (apply min))
+                   (* 1.05 (->> data :values (map first) (apply max)))] ;; add 5% to max just to leave a small gap
+          y-range [(min (->> data :values (map second) (apply min)) (-> reg-line first second))
+                   (* 1.05 (max (->> data :values (map second) (apply max)) (-> reg-line second second)))]
+          x-major (.floor js/Math (/ (second x-range) 10))
+          x-minor (/ x-major 2)
+          y-major (.floor js/Math (/ (second y-range) 10))
+          y-minor (/ y-major 2)]
+      {:data data
+       :x-range x-range
+       :y-range y-range
+       :x-minor x-minor
+       :x-major x-major
+       :y-minor y-minor
+       :y-major y-major})))
+
+(defn load-xyplot-data [key data-chan xyplot-data resp]
+  (om/update! xyplot-data [:error] false)
+  (let [state (-> @(om/update! xyplot-data key (:measurements resp)) :properties :xyplot)]
```

Comment:
Yes, om/update! returns a cursor to the whole app-state. This is left over from some experimentation, do you want me to tidy it up?